content/guides/autodiscovery.md

```diff
@@ -4,37 +4,75 @@ kind: guide
+Second, because templates don't identify specific instances of a monitored service—which host? which port?—Autodiscovery needs one or more **service identifiers** for each template so it can find real values for the template variables. For Docker, service identifiers are [image names or container labels](#service-identifiers). With identifiers in hand, Autodiscovery can figure out which IP(s) and port(s) to substitute into the template.
+
+Finally, Autodiscovery can load check templates from places other than disk. Other possible **template sources** include key-value (KV) stores like Consul, and, when running on Kubernetes, pod annotations.
+
+### Different Execution
+
+When the Agent starts with Autodiscovery enabled, it loads check templates from all available template sources (not just one or the other). But unlike in a traditional Agent setup, it doesn't run all checks all the time; it decides which checks to enable given the containers that are currently running.
+
+As the Agent inspects each template and its service identifiers, it builds a mapping of service identifiers to template YAML (or JSON). Then, it inspects every running Docker container on the Agent's host, looking for containers that match any of the service identifiers (keys) in the identifier-to-template mapping. For each match, the Agent generates a static check configuration by substituting the matching container's IP address and port into the corresponding template. Then it enables the check.
+
+The Agent watches for Docker events—container creation, destruction, starts, and stops—and recompiles check configurations from the templates on such events. Any time Autodiscovery finds another instance of the some service it had previously generated configuration for—whether it's an additional container or the previous one redeployed—it builds and enables another static configuration.
+
+There are a few caveats:
+
+- The Agent caches templates it gets from KV stores. Changes to KV store templates require a restart of the Agent.
+- something about precedence of templates when multiple sources provided?
+- others?
+
+---
+
+The next section clarifies this abstract description of Autodiscovery with some examples.
+
+# How to set it up
+
+No matter what container orchestration platform you use, you'll first need to run a single [docker-dd-agent container](https://hub.docker.com/r/datadog/docker-dd-agent/) on every host in your cluster. Then you'll write check templates, storing them in one or more template sources.
+
+## Running the Agent Container
+
+If you use Kubernetes, see the [Kubernetes integration page](http://docs.datadoghq.com/integrations/kubernetes/#installation) for instructions on running docker-dd-agent. If you use Amazon ECS, see [its integration page](http://docs.datadoghq.com/integrations/ecs/#installation).
+
+If you use Docker Swarm, run the following command on one of your manager nodes:
 
     docker service create \
```

Comment:
shouldnt we be consistent here? if we tell folks to look elsewhere for k8s and ecs, can't we do the same for swarm?